contracts/contracts/vault/VaultCore.sol

```diff
@@ -165,20 +165,21 @@ contract VaultCore is VaultStorage {
         emit Redeem(msg.sender, _amount);
 
         // Send outputs
-        for (uint256 i = 0; i < allAssets.length; i++) {
+        uint256 assetCount = allAssets.length;
+        for (uint256 i = 0; i < assetCount; ++i) {
```

Comment:
oh right this is because the for loop is checking for the end condition of each loop that it is performing?

Reply:
correct. `allAssets.length` will do an SLOAD in each loop